### PR TITLE
handle responses with `ok: false`

### PR DESCRIPTION
**Pull request by afeld**
In trying to debug https://github.com/rauchg/slackin/issues/38, finally found that there was an error condition that wasn't being handled:
 
screen shot 2015-04-27 at 4 30 11 pm

This PR:

* Handles a JSON response from the Slack invite API where the `ok` property is `false`
* Fixes passing of error messages to client
* Adds tests

/cc @ultrasaurus

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,12 +15,15 @@
     "vd": "0.1.0"
   },
   "devDependencies": {
-    "browserify": "8.1.1",
-    "mocha": "2.1.0",
-    "supertest": "0.15.0"
+    "mocha": "^2.2.4",
+    "nock": "^0.48.1",
+    "supertest": "^0.15.0"
```
Comment (rauchg) [inline package.json]: can you please pin?

**Comment by rauchg** [comment]
This is super cool!

File: package.json (hunk 1 above)
Comment (afeld) [inline package.json]: Sorry, happened automatically with `npm install nock --save-dev`...didn't even notice.

**Comment by afeld** [comment]
Updated!

**Comment by rauchg** [comment]
Can't merge 
image

**Comment by afeld** [comment]
Updated!

**Pull request merged**